webpack/assets/javascripts/react_app/common/hooks/Legacy/LegacyLoaderHook.js

```diff
@@ -0,0 +1,60 @@
+import { useState, useEffect } from 'react';
+import { STATUS } from '../../../constants';
+import { removeElements } from './helpers';
+import { getApiResponse } from '../../../redux/API/APIHelpers';
+
+/**
+ * A custom hook that returns a legacy conent to be wrapped in a react component
+ * @param  {string} type the  method for the ajax request (i.e 'post', 'get' etc)
+ * @param  {string} url the url for the ajax request
+ * @param  {object} options includes request's headers and params, and DOM maniplation helpers
+ * @return {object} returns an object that contains the request's status and the parsed html to renderd in 'dangerouslySetInnerHTML'
+ */
+
+export const useLegacyLoader = (
```

Comment:
this hook creates an ajax request for loading legacy content wrapped with react component.